crates/subspace-farmer/src/plotting.rs

```diff
@@ -72,32 +65,50 @@ impl FarmerData {
 /// Assumes `plot`, `commitment`, `object_mappings`, `client` and `identity` are already initialized
 impl Plotting {
     /// Returns an instance of plotting, and also starts a concurrent background plotting task
-    pub fn start<T: RpcClient + Clone + Send + Sync + 'static>(
+    pub async fn start<T: RpcClient + Clone + Send + Sync + 'static>(
```

Comment:
This doesn't need to be async. With `spawn_listening_to_blocks` changed to not do internal spawning it doesn't need to be awaited, externalizing `client.farmer_metadata()` query and removing useless (in this context) spawning around `plot.get_last_root_block()`, start function will remain synchronous like before.

UPD: Hm, I see it is no longer async in the later commit.